src/http-handler.js

```diff
@@ -0,0 +1,43 @@
+import window from 'global/window';
+
+export const httpResponseHandler = (callback) => (err, response, responseBody) => {
+  // if the XHR failed, return that error
+  if (err) {
+    callback(err);
+    return;
+  }
+
+  // if the HTTP status code is 4xx or 5xx, the request also failed
+  if (response.statusCode >= 400 && response.statusCode <= 599) {
+    let cause;
+
+    if (window.TextDecoder) {
```

Comment:
Does any of this throw if responseBody is empty?

Reply:
Might be worth adding some tests for these two functions.

Reply:
Yup, definitely need a bunch of tests before this is ready.
Looks like decoding nothing doesn't throw.
```js
> var td = new TextDecoder('utf-8')
> td.decode()
< ""
```

Reply:
Moving to https://github.com/videojs/xhr/pull/7, I did end up wrapping this in a try catch just in case.